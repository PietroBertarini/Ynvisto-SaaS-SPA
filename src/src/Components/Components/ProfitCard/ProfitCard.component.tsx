import React from 'react';
import Typography from '@material-ui/core/Typography';
import {
  Card, CardActionArea,
} from '@material-ui/core';
import { Link } from 'react-router-dom';
import Box from '@material-ui/core/Box';
import CardContent from '@material-ui/core/CardContent';
import Grid from '@material-ui/core/Grid';
import { useStyles } from './ProfitCard.styles';

export enum EProfitCardStatus {
    PAID = 'PAID',
    NO_DARF = 'NO_DARF',
    PENDING = 'PENDING',
    LATE = 'LATE',
    NOT_FINISHED = 'NOT_FINISHED',
}

export interface ProfitCardProps {
    /**
     * Month of card
     */
    month: string;
    /**
     * Amount of sell volume of month
     */
    profit : number;
    /**
     * Amount of tax volume
     */
    tax: number
    /**
     * Actual status of profitCard
     */
    status: EProfitCardStatus;

}

/**
 * Monthly Card of Profit and Tax Page
 * @constructor
 */
function ProfitCard(props: ProfitCardProps) {
  const classes = useStyles(props);
  const {
    profit, tax, month,
  } = props;

  return (
    <Card className={classes.root}>
      <Link to={`${month}`}>
        <CardActionArea className={classes.actionArea}>
          <CardContent>
            <Box display="flex">
              <Typography className={classes.month}>
                {month}
              </Typography>
            </Box>
            <Grid container direction="row" className={classes.displayFlex}>
              <Grid item xs={7} className={classes.displayFlex}>
                <Box
                  display="flex"
                  flexDirection="column"
                  alignItems="center"
                  justifyContent="flex-start"
                  className={classes.box}
                  alignSelf="center"
                >
                  {profit >= 0 && (
                  <Typography className={classes.profitPositive}>
                    {profit.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
                  </Typography>
                  )}
                  {profit < 0 && (
                    <Typography className={classes.profitNegative}>
                      {profit.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
                    </Typography>
                  )}
                  <Typography className={classes.subtitle}>
                    {profit >= 0 ? 'Lucro' : 'Preju√≠zo'}
                  </Typography>
                </Box>
              </Grid>
              <Grid item xs={5} className={classes.displayFlex}>
                <Box
                  display="flex"
                  flexDirection="column"
                  alignItems="center"
                  justifyContent="flex-start"
                  className={classes.box}
                  alignSelf="center"
                >
                  <Typography className={classes.tax}>
                    {tax.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
                  </Typography>
                  <Typography className={classes.subtitle}>
                    Imposto
                  </Typography>
                </Box>
              </Grid>
            </Grid>
          </CardContent>
        </CardActionArea>
      </Link>
    </Card>
  );
}
export default ProfitCard;
