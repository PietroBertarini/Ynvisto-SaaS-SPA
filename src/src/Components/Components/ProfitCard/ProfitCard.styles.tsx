import { makeStyles, createStyles } from '@material-ui/core/styles';

const styledBy = (property, mapping) => (props) => mapping[props[property]];

export const useStyles = makeStyles(() => createStyles({
  root: {
    width: '100%',
    height: '100%',
    minWidth: 222,
    minHeight: 90,
    borderRadius: '2.5vh',
    background: styledBy('status', {
      PAID: 'linear-gradient(235.08deg, #05C0E9 -83%, #32323E 38.57%)',
      NO_DARF: 'linear-gradient(235.08deg, #05E985 -83%, #236D55 12.15%, #32323E 38.57%);',
      PENDING: 'linear-gradient(235.08deg, #E9C46A -83%, #32323E 38.57%)',
      LATE: 'linear-gradient(235.08deg, #EB5050 -83%, #32323E 38.04%)',
      NOT_FINISHED: 'linear-gradient(235.08deg, #FFFFFF -83%, #32323E 38.57%)',
    }),
  },
  actionArea: {
    display: 'inline-block',
  },
  profitPositive: {
    fontSize: '1.6vw',
    width: 'auto',
    color: styledBy('status', {
      PAID: '#05C0E9',
      NO_DARF: '#05E985',
      PENDING: '#E9C46A',
      LATE: '#EB5050',
      NOT_FINISHED: '#B1B1B1',
    }),
  },
  profitNegative: {
    fontSize: '1.6vw',
    width: 'auto',
    color: styledBy('status', {
      PAID: '#EB5050',
      NO_DARF: '#EB5050',
      PENDING: '#E9C46A',
      LATE: '#EB5050',
      NOT_FINISHED: '#B1B1B1',
    }),
  },

  tax: {
    fontSize: '1.4vw',
    width: 'auto',
    color: styledBy('status', {
      PAID: '#05C0E9',
      NO_DARF: '#05E985',
      PENDING: '#E9C46A',
      LATE: '#EB5050',
      NOT_FINISHED: '#B1B1B1',
    }),
  },
  impostoBox: {
    display: 'flex',
  },
  subtitle: {
    fontSize: '1vw',
    color: '#8E8E94',
  },
  month: {
    fontSize: '1.2vw',
    color: '#FFFFFF',
  },
  box: {
    width: 'auto',
  },
  displayFlex: {
    display: 'flex',
  },

}));
