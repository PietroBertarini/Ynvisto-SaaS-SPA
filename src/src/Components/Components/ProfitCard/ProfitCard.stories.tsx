import React from 'react';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Meta, Story } from '@storybook/react/types-6-0';
import ProfitCard, { EProfitCardStatus, ProfitCardProps } from './ProfitCard.component';

export default {
  title: 'Components/ProfitCard',
  component: ProfitCard,
} as Meta;

const Template: Story<ProfitCardProps> = (args) => <ProfitCard {...args} />;

export const StatusLate = Template.bind({});
StatusLate.args = {
  profit: 30_304.94,
  tax: 506.14,
  status: EProfitCardStatus.LATE,
  month: 'Maio',
};
export const StatusInProgress = Template.bind({});
StatusInProgress.args = {
  profit: 30_304.94,
  tax: 506.14,
  status: EProfitCardStatus.NOT_FINISHED,
  month: 'Maio',
};
export const StatusNoDarf = Template.bind({});
StatusNoDarf.args = {
  profit: 30_304.94,
  tax: 506.14,
  status: EProfitCardStatus.NO_DARF,
  month: 'Maio',
};
export const StatusPaid = Template.bind({});
StatusPaid.args = {
  profit: 30_304.94,
  tax: 506.14,
  status: EProfitCardStatus.PAID,
  month: 'Maio',
};
export const StatusPending = Template.bind({});
StatusPending.args = {
  profit: 30_304.94,
  tax: 506.14,
  status: EProfitCardStatus.PENDING,
  month: 'Maio',
};
