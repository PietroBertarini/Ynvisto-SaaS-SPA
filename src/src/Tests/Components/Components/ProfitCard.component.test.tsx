import React from 'react';
import { shallow } from 'enzyme';
import ProfitCard, { EProfitCardStatus, ProfitCardProps } from '../../../Components/Components/ProfitCard/ProfitCard.component';

test('Renders ProfitCardProps components and match to previously snapshot', () => {
  const mockProps : ProfitCardProps = {
    profit: 30_304.94,
    tax: 506.14,
    month: 'Maio',
    status: EProfitCardStatus.PENDING,
  };

  const paidStatusComponent = shallow(
    <ProfitCard
      status={EProfitCardStatus.PAID}
      month={mockProps.month}
      profit={mockProps.profit}
      tax={mockProps.tax}
    />,
  );
  const pendentStatusComponent = shallow(
    <ProfitCard
      status={EProfitCardStatus.PENDING}
      month={mockProps.month}
      profit={mockProps.profit}
      tax={mockProps.tax}
    />,
  );
  const NoDarfStatusComponent = shallow(
    <ProfitCard
      status={EProfitCardStatus.NO_DARF}
      month={mockProps.month}
      profit={mockProps.profit}
      tax={mockProps.tax}
    />,
  );
  const inProgressStatusComponent = shallow(
    <ProfitCard
      status={EProfitCardStatus.NOT_FINISHED}
      month={mockProps.month}
      profit={mockProps.profit}
      tax={mockProps.tax}
    />,
  );
  const lateStatusComponent = shallow(
    <ProfitCard
      status={EProfitCardStatus.LATE}
      month={mockProps.month}
      profit={mockProps.profit}
      tax={mockProps.tax}
    />,
  );
  expect(paidStatusComponent).toMatchSnapshot();
  expect(pendentStatusComponent).toMatchSnapshot();
  expect(NoDarfStatusComponent).toMatchSnapshot();
  expect(inProgressStatusComponent).toMatchSnapshot();
  expect(lateStatusComponent).toMatchSnapshot();
});
