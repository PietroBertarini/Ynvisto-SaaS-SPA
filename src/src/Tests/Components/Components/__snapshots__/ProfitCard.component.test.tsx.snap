// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ProfitCardProps components and match to previously snapshot 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1 makeStyles-root-11"
>
  <Link
    to="Maio"
  >
    <WithStyles(ForwardRef(CardActionArea))
      className="makeStyles-actionArea-2"
    >
      <WithStyles(ForwardRef(CardContent))>
        <Styled(MuiBox)
          display="flex"
        >
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-month-8"
          >
            Maio
          </WithStyles(ForwardRef(Typography))>
        </Styled(MuiBox)>
        <WithStyles(ForwardRef(Grid))
          className="makeStyles-displayFlex-10"
          container={true}
          direction="row"
        >
          <WithStyles(ForwardRef(Grid))
            className="makeStyles-displayFlex-10"
            item={true}
            xs={7}
          >
            <Styled(MuiBox)
              alignItems="center"
              alignSelf="center"
              className="makeStyles-box-9"
              display="flex"
              flexDirection="column"
              justifyContent="flex-start"
            >
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-profitPositive-3 makeStyles-profitPositive-12"
              >
                R$ 30.304,94
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-subtitle-7"
              >
                Lucro
              </WithStyles(ForwardRef(Typography))>
            </Styled(MuiBox)>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            className="makeStyles-displayFlex-10"
            item={true}
            xs={5}
          >
            <Styled(MuiBox)
              alignItems="center"
              alignSelf="center"
              className="makeStyles-box-9"
              display="flex"
              flexDirection="column"
              justifyContent="flex-start"
            >
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-tax-5 makeStyles-tax-14"
              >
                R$ 506,14
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-subtitle-7"
              >
                Imposto
              </WithStyles(ForwardRef(Typography))>
            </Styled(MuiBox)>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(CardContent))>
    </WithStyles(ForwardRef(CardActionArea))>
  </Link>
</WithStyles(ForwardRef(Card))>
`;

exports[`Renders ProfitCardProps components and match to previously snapshot 2`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1 makeStyles-root-15"
>
  <Link
    to="Maio"
  >
    <WithStyles(ForwardRef(CardActionArea))
      className="makeStyles-actionArea-2"
    >
      <WithStyles(ForwardRef(CardContent))>
        <Styled(MuiBox)
          display="flex"
        >
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-month-8"
          >
            Maio
          </WithStyles(ForwardRef(Typography))>
        </Styled(MuiBox)>
        <WithStyles(ForwardRef(Grid))
          className="makeStyles-displayFlex-10"
          container={true}
          direction="row"
        >
          <WithStyles(ForwardRef(Grid))
            className="makeStyles-displayFlex-10"
            item={true}
            xs={7}
          >
            <Styled(MuiBox)
              alignItems="center"
              alignSelf="center"
              className="makeStyles-box-9"
              display="flex"
              flexDirection="column"
              justifyContent="flex-start"
            >
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-profitPositive-3 makeStyles-profitPositive-16"
              >
                R$ 30.304,94
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-subtitle-7"
              >
                Lucro
              </WithStyles(ForwardRef(Typography))>
            </Styled(MuiBox)>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            className="makeStyles-displayFlex-10"
            item={true}
            xs={5}
          >
            <Styled(MuiBox)
              alignItems="center"
              alignSelf="center"
              className="makeStyles-box-9"
              display="flex"
              flexDirection="column"
              justifyContent="flex-start"
            >
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-tax-5 makeStyles-tax-18"
              >
                R$ 506,14
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-subtitle-7"
              >
                Imposto
              </WithStyles(ForwardRef(Typography))>
            </Styled(MuiBox)>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(CardContent))>
    </WithStyles(ForwardRef(CardActionArea))>
  </Link>
</WithStyles(ForwardRef(Card))>
`;

exports[`Renders ProfitCardProps components and match to previously snapshot 3`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1 makeStyles-root-19"
>
  <Link
    to="Maio"
  >
    <WithStyles(ForwardRef(CardActionArea))
      className="makeStyles-actionArea-2"
    >
      <WithStyles(ForwardRef(CardContent))>
        <Styled(MuiBox)
          display="flex"
        >
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-month-8"
          >
            Maio
          </WithStyles(ForwardRef(Typography))>
        </Styled(MuiBox)>
        <WithStyles(ForwardRef(Grid))
          className="makeStyles-displayFlex-10"
          container={true}
          direction="row"
        >
          <WithStyles(ForwardRef(Grid))
            className="makeStyles-displayFlex-10"
            item={true}
            xs={7}
          >
            <Styled(MuiBox)
              alignItems="center"
              alignSelf="center"
              className="makeStyles-box-9"
              display="flex"
              flexDirection="column"
              justifyContent="flex-start"
            >
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-profitPositive-3 makeStyles-profitPositive-20"
              >
                R$ 30.304,94
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-subtitle-7"
              >
                Lucro
              </WithStyles(ForwardRef(Typography))>
            </Styled(MuiBox)>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            className="makeStyles-displayFlex-10"
            item={true}
            xs={5}
          >
            <Styled(MuiBox)
              alignItems="center"
              alignSelf="center"
              className="makeStyles-box-9"
              display="flex"
              flexDirection="column"
              justifyContent="flex-start"
            >
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-tax-5 makeStyles-tax-22"
              >
                R$ 506,14
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-subtitle-7"
              >
                Imposto
              </WithStyles(ForwardRef(Typography))>
            </Styled(MuiBox)>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(CardContent))>
    </WithStyles(ForwardRef(CardActionArea))>
  </Link>
</WithStyles(ForwardRef(Card))>
`;

exports[`Renders ProfitCardProps components and match to previously snapshot 4`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1 makeStyles-root-23"
>
  <Link
    to="Maio"
  >
    <WithStyles(ForwardRef(CardActionArea))
      className="makeStyles-actionArea-2"
    >
      <WithStyles(ForwardRef(CardContent))>
        <Styled(MuiBox)
          display="flex"
        >
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-month-8"
          >
            Maio
          </WithStyles(ForwardRef(Typography))>
        </Styled(MuiBox)>
        <WithStyles(ForwardRef(Grid))
          className="makeStyles-displayFlex-10"
          container={true}
          direction="row"
        >
          <WithStyles(ForwardRef(Grid))
            className="makeStyles-displayFlex-10"
            item={true}
            xs={7}
          >
            <Styled(MuiBox)
              alignItems="center"
              alignSelf="center"
              className="makeStyles-box-9"
              display="flex"
              flexDirection="column"
              justifyContent="flex-start"
            >
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-profitPositive-3 makeStyles-profitPositive-24"
              >
                R$ 30.304,94
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-subtitle-7"
              >
                Lucro
              </WithStyles(ForwardRef(Typography))>
            </Styled(MuiBox)>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            className="makeStyles-displayFlex-10"
            item={true}
            xs={5}
          >
            <Styled(MuiBox)
              alignItems="center"
              alignSelf="center"
              className="makeStyles-box-9"
              display="flex"
              flexDirection="column"
              justifyContent="flex-start"
            >
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-tax-5 makeStyles-tax-26"
              >
                R$ 506,14
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-subtitle-7"
              >
                Imposto
              </WithStyles(ForwardRef(Typography))>
            </Styled(MuiBox)>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(CardContent))>
    </WithStyles(ForwardRef(CardActionArea))>
  </Link>
</WithStyles(ForwardRef(Card))>
`;

exports[`Renders ProfitCardProps components and match to previously snapshot 5`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1 makeStyles-root-27"
>
  <Link
    to="Maio"
  >
    <WithStyles(ForwardRef(CardActionArea))
      className="makeStyles-actionArea-2"
    >
      <WithStyles(ForwardRef(CardContent))>
        <Styled(MuiBox)
          display="flex"
        >
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-month-8"
          >
            Maio
          </WithStyles(ForwardRef(Typography))>
        </Styled(MuiBox)>
        <WithStyles(ForwardRef(Grid))
          className="makeStyles-displayFlex-10"
          container={true}
          direction="row"
        >
          <WithStyles(ForwardRef(Grid))
            className="makeStyles-displayFlex-10"
            item={true}
            xs={7}
          >
            <Styled(MuiBox)
              alignItems="center"
              alignSelf="center"
              className="makeStyles-box-9"
              display="flex"
              flexDirection="column"
              justifyContent="flex-start"
            >
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-profitPositive-3 makeStyles-profitPositive-28"
              >
                R$ 30.304,94
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-subtitle-7"
              >
                Lucro
              </WithStyles(ForwardRef(Typography))>
            </Styled(MuiBox)>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            className="makeStyles-displayFlex-10"
            item={true}
            xs={5}
          >
            <Styled(MuiBox)
              alignItems="center"
              alignSelf="center"
              className="makeStyles-box-9"
              display="flex"
              flexDirection="column"
              justifyContent="flex-start"
            >
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-tax-5 makeStyles-tax-30"
              >
                R$ 506,14
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-subtitle-7"
              >
                Imposto
              </WithStyles(ForwardRef(Typography))>
            </Styled(MuiBox)>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(CardContent))>
    </WithStyles(ForwardRef(CardActionArea))>
  </Link>
</WithStyles(ForwardRef(Card))>
`;
